idp-service/src/main/java/io/mosip/idp/services/LoggerAuditService.java

```diff
@@ -0,0 +1,50 @@
+package io.mosip.idp.services;
+
+import io.mosip.idp.core.dto.AuditableIdPTransaction;
+import io.mosip.idp.core.spi.AuditWrapper;
+import io.mosip.idp.core.util.IdPAction;
+import lombok.extern.slf4j.Slf4j;
+import org.slf4j.MDC;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.scheduling.annotation.Async;
+import org.springframework.stereotype.Component;
+
+import javax.validation.constraints.NotNull;
+
+@ConditionalOnProperty(value = "mosip.idp.audit.wrapper.impl", havingValue = "LoggerAuditService")
+@Component
+@Slf4j
+public class LoggerAuditService implements AuditWrapper {
+
+    @Async
+    @Override
+    public void logAudit(@NotNull IdPAction action, @NotNull AuditableIdPTransaction transaction, Throwable t) {
+        addTransactionDetailsToMDC(transaction);
```

Comment:
As discussed lets add the status seperately

Reply:
Done